### PR TITLE
MM-14273 Add boolean element support to interactive dialog

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add boolean element support to interactive dialog

While testing https://mattermost.atlassian.net/browse/MM-14273, I found that no corresponding PR for mobile app. 

#### Ticket Link
Jira ticket: [MM-14273](https://mattermost.atlassian.net/browse/MM-14273)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on: [iOS and Android simulator/emulator] 

#### Screenshots
iOS:
Screen Shot 2019-09-11 at 3 05 05 AM
Screen Shot 2019-09-11 at 3 05 22 AM
Screen Shot 2019-09-11 at 3 06 28 AM

Android:
Screen Shot 2019-09-11 at 3 14 45 AM
Screen Shot 2019-09-11 at 3 14 59 AM
Screen Shot 2019-09-11 at 3 15 38 AM

### CODE REVIEW

**Comment by saturninoabril** [comment]
@asaadmahmood I don't see a design for this.  Could you please review on UX?  Thanks

File: app/screens/interactive_dialog/index.js [hunk 1]
```diff
@@ -20,6 +21,7 @@ function mapStateToProps(state) {
         elements: data.dialog.elements,
         title: data.dialog.title,
         iconUrl: data.dialog.icon_url,
+        isLandscape: isLandscape(state),
```
Comment (enahum) [inline app/screens/interactive_dialog/index.js]: consider passing this value from the parent component, and if not available consider calculating it by using `Dimensions.get('window')`.

The reason behind it is to make it simpler when migrating to realm as the state for landscape won't be available anymore.

Comment (saturninoabril) [reply app/screens/interactive_dialog/index.js]: Okay, did the latter by computing from `Dimensions`.  Also add listener for dimension/orientation change.

File: app/screens/interactive_dialog/interactive_dialog.js [hunk 2]
```diff
@@ -47,12 +47,27 @@ export default class InteractiveDialog extends PureComponent {
...
             values,
             errors: {},
+            isLandscape: this.isLandscape(),
             submitting: false,
         };
     }
 
     componentDidMount() {
         this.navigationEventListener = Navigation.events().bindComponent(this);
+        Dimensions.addEventListener('change', this.orientationDidChange);
```
Comment (enahum) [inline app/screens/interactive_dialog/interactive_dialog.js]: Is the listener required? Getting the dimension in the render method, at times is all you need. Try it out, if it works (I’m 99% sure) you can remove the listener

Comment (saturninoabril) [reply app/screens/interactive_dialog/interactive_dialog.js]: Yes, it's needed.  I tried first on the render method and works fine on current orientation but didn't update on orientation change.

**Comment by enahum** [comment]
@saturninoabril please merge master

**Comment by saturninoabril** [comment]
Thanks, done merge and fixed conflict.

**Pull request merged**